packages/shell-api/src/collection.ts

```diff
@@ -2056,4 +2056,66 @@ export default class Collection extends ShellApiWithMongoClass {
       ...options
     });
   }
+
+  @serverVersions(['6.0.0', ServerVersions.latest])
+  @returnsPromise
+  @apiVersions([])
+  async getSearchIndexes(options?: any): Promise<Document[]> {
+    this._emitCollectionApiCall('getSearchIndexes', { options });
+    return await this._mongo._serviceProvider.getSearchIndexes(
+      this._database._name,
+      this._name,
+      { ...await this._database._baseOptions(), ...options }
+    );
+  }
+
+  @serverVersions(['6.0.0', ServerVersions.latest])
+  @returnsPromise
+  @apiVersions([])
+  async createSearchIndex(description: any): Promise<string> {
+    this._emitCollectionApiCall('createSearchIndex', { description });
+    const results = await this._mongo._serviceProvider.createSearchIndexes(
+      this._database._name,
+      this._name,
+      [description]
+    );
+    return results[0];
+  }
+
+  @serverVersions(['6.0.0', ServerVersions.latest])
+  @returnsPromise
+  @apiVersions([])
+  async createSearchIndexes(descriptions: any[]): Promise<string[]> {
```

Comment:
```suggestion
  async createSearchIndexes(descriptions: Document[]): Promise<string[]> {
```

Reply:
I had it as Document, but then deliberately changed it to any because it felt like that's more likely to stand out as something we have to type properly once we can.

Reply:
See https://github.com/mongodb-js/mongosh/pull/1477/commits/de6038f8582c56198e30dbec9606a8a97739e9d5

Reply:
We’re using `Document` quite extensively here in the shell-api package, and that’s probably a good thing, because these methods are user-facing and it’s not like we could tell the user to really always give us input that exactly matches the driver types